aws/adapter.go

```diff
@@ -626,13 +628,29 @@ func (a *Adapter) SecurityGroupID() string {
 // FindManagedStacks returns all CloudFormation stacks containing the controller management tags
 // that match the current cluster and are ready to be used. The stack status is used to filter.
 func (a *Adapter) FindManagedStacks() ([]*Stack, error) {
-	stacks, err := findManagedStacks(a.cloudformation, a.ClusterID(), a.controllerID)
+	stacks, err := findManagedStacks(a.cloudformation, a.ClusterID(), a.controllerID, a.stackLastTargerGroupARNs)
 	if err != nil {
 		return nil, err
 	}
 	return stacks, nil
 }
 
+func (a *Adapter) UpdateStackLastTargetGroupARNs(stack *Stack) {
+	if _, ok := a.stackLastTargerGroupARNs[stack.Name]; !ok {
+		if len(stack.TargetGroupARNs) > 0 {
```

Comment:
why do you check this here?

Reply:
The check can be moved outside the function yes, I just thought it made sense here.